### PR TITLE
Spectral Grid generation & visualization

### PR DESCRIPTION
**Pull request by Nimishkhurana**
Added Jupyter Notebook which plots spectra using a jupyter widget's slider for grid.teff parameter.

### CODE REVIEW

**Comment by dchu808** [comment]
Great start. It would be useful to make widget sliders for the other parameters as well (such as logg)

**Comment by wkerzendorf** [comment]
as @dchu808 said - please include more parameters.

**Comment by Nimishkhurana** [comment]
Ok @dchu808 @wkerzendorf  I will do it today.

**Comment by Nimishkhurana** [comment]
I have made widgets for logg and mh also.Please review @wkerzendorf

**Comment by wkerzendorf** [comment]
I'll leave you in the capable hands of @dchu808

File: Starkit_plot_spectra_slider.ipynb [hunk 1]
```diff
@@ -104,41 +68,50 @@
...
    "source": [
-    "from ipywidgets import interact\n",
-    "import ipywidgets as widgets"
+    "# Generate a slider for all parameters of spectra\n",
+    "def f(teff = 4000,logg = 3,mh = 0):\n",
+    "    grid.teff = teff\n",
+    "    grid.logg = logg\n",
+    "    grid.mh = mh\n",
+    "    wave,flux = grid()\n",
+    "    plt.title(\"Spectrum at Teff = {}  logg={}  mh={}\".format(teff,logg,mh))\n",
```
Comment (dchu808) [inline Starkit_plot_spectra_slider.ipynb]: It's helpful seeing the parameters in the title, good idea

**Comment by Nimishkhurana** [comment]
Ok @dchu808 will do it.

**Comment by Nimishkhurana** [comment]
I have implemented the second point you said i.e making 2 spectrum simultaneously. But to make the matplotlib plot  interactive ,I think there is only one way : `%matplotlib notebook ` and this doesn't seems to work very wll with ipython's `interact`. 
The plots work great with sliders if they are not plotted in with interactive backened.Please review once after commenting out `%matplotlib notebook`.

My suggestion is to try other visualization libraries like seaborn,bokeh or plotly .This would be great, but if we are ready to add more dependency.Please give your opinion after reviewing.

**Comment by wkerzendorf** [comment]
you can definitely use other vizualization libraries

File: Starkit_plot_spectra_slider.ipynb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    "def plot_spectra(teff1 = 4000,logg1 = 3,mh1 = 0,teff2 = 4000,logg2 = 3,mh2 = 0):\n",
+    "    grid.teff = teff1\n",
+    "    grid.logg = logg1\n",
+    "    grid.mh = mh1\n",
+    "    wave1,flux1 = grid()\n",
+    "    grid.teff = teff2\n",
+    "    grid.logg = logg2\n",
+    "    grid.mh = mh2\n",
+    "    wave2,flux2 = grid()\n",
+    "    plt.plot(wave1,flux1,'r',label = 'Teff = {}  logg={}  mh={}'.format(teff1,logg1,mh1))\n",
```
Comment (dchu808) [inline Starkit_plot_spectra_slider.ipynb]: I recommend not using red and green as the 2 colors for colorblindness

File: Starkit_plot_spectra_slider.ipynb [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    "    plt.plot(wave2,flux2,'g',label = 'Teff = {}  logg={}  mh={}'.format(teff2,logg2,mh2))\n",
+    "    plt.legend()\n",
```
Comment (dchu808) [inline Starkit_plot_spectra_slider.ipynb]: Legend is a good way to show the comparison, great!

File: Starkit_plot_spectra_slider.ipynb [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    "    plt.show()\n"
```
Comment (dchu808) [inline Starkit_plot_spectra_slider.ipynb]: Add labels for the axes

**Comment by Nimishkhurana** [comment]
I will do the changes you recommended @dchu808.

**Comment by Nimishkhurana** [comment]
Also i would like to know what are specific things more to do in this project in GSoC.

**Comment by Nimishkhurana** [comment]
@dchu808 I have changed plot colors and labelled the axes.
Now the plots look like this:-
screenshot from 2019-02-28 00-13-50

I will aslo try other vizualisation libraries shortly.

**Comment by Nimishkhurana** [comment]
@dchu808 @wkerzendorf Can you please guide what more need to be done in the project. What other visualization and UI is expected. With complete information i will be able to make plan and good proposal for GSoC.

**Comment by dchu808** [comment]
@Nimishkhurana Have you looked into some of the visualization libraries as discussed before? Another potential thing would be to incorporate a way to visualize differences between grids. You can also start to put together the application, it can just be a skeleton of an application. Brainstorm what you can do for a summer project, you may come up with something the rest of us haven't thought about.

**Comment by Nimishkhurana** [comment]
@dchu808 I have looked into some of visualization libraries but I was busy because of university exams so  couldn't visualize grids in them. I will try them in this week and also think what can be done more this summer for grid vizualization.

**Comment by urstrulykkr** [comment]
@dchu808 I would like to contribute to this task. Also would like to use the above guidance given.
Will soon PR my task :)

**Comment by Nimishkhurana** [comment]
Made a dashboard using DASH which is required in dependencies, so Travis tests failed.
They can be pip installed by:

`pip install dash==0.39.0`
`pip install dash-daq==0.1.0`

Screenshot from 2019-03-31 23-43-34

To run the Dashboard in localhost:

`python app.py`

This is just the preliminary work to show the basic idea for GSoC. More functanilities will be added in summer(like complete functionality for comparing more than one grids, adding plugin support and UI will be improved.

**Pull request closed without merging**